docstore/docstore.go

```diff
@@ -28,3 +29,16 @@ type DocStore interface {
 	Getter
 	Putter
 }
+
+var validKeyRegexp = regexp.MustCompile(`^[a-zA-Z0-9_-]*$`)
```

Comment:
Seems like we're leaving out a lot of characters here (e.g. for S3: https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html).  Don't you need at least `.` for file extensions?  Also, are we saying subfolders are completely forbidden?  Not sure how docstore has been used before, but that seems potentially a valid use case.